documentation/v&v/V&V.tex

```diff
@@ -0,0 +1,588 @@
+			\item Search for books that they would like to purchase.
+		\ei
+	\item A system that allows buyers and sellers to privately contact each other, agree on a price, and choose where they would like to meet.
+	\item The application will be accessible from any device with a modern browser (phone, tablet, desktop, etc.) to provide the best experience.
+\ee
+
+
+
+\section{Background}
+\label{sec:background}
+University students often like to buy/sell used textbooks. Sometimes it's for the notes written in the margins, but often it's because new textbooks are quite expensive.\\
+Currently at McMaster University, students use online websites like Facebook (\url{https://www.facebook.com/groups/macusedtexts}) or the campus store's buyback program (\url{https://campusstore.mcmaster.ca/information/guaranteed-buyback.html}, suspended because of COVID) to sell their books. These current solutions are either disorganized or take a cut of the transaction. Book Bazaar aims to create an organized marketplace that allows users to sell books for a better price than at the campus store.\\
+The project, Book Bazar is a website that addresses these issues. McMaster community members may buy/sell textbooks with ease using Book Bazar.
+\section{Roadmap}
+\label{sec: roadmap}
+Using the milestones, listed in Section 6 of the Development Process document, the team aims to have Book Bazar completed by the week of March 6, a week prior to the team's f\textbf{}inal presentation.
+
+\section{Testing Plan}
+Integration testing was used for all top-level modules which transitively tests all bottom level modules. This style of testing was chosen because it effectively tests how users will interact with the system instead of implementation details. Few unit tests were written because, in practice, they are often burdensome to write and maintain while providing minimal benefit in many cases due to much of the system being mocked out.
+
+\subsection{Testing Tools}
+The team will use the following tools for implementing tests:\\
+\bi
+    \item Cypress -- used to run all integration tests (API and UI).
+    \item Jest -- used for writing unit tests.
+    \item Mock Service Workers -- used for mocking out external service APIs.
+\ei
+\subsection{API Tests}
+In the design specification the behaviour of HTTP handler modules was defined. This behaviour was used to define an expected output for a variety of inputs. This required creating tooling to build and teardown a testing database.
+
+A serverless architecture was used for API execution which means that horizontal scaling of the system is handled by \href{https://vercel.com}{Vercel}, the hosting provider. Therefore, \textbf{load testing} was not performed.
+
+
+\subsection{UI Tests}
+The team is testing with the assumption that the user uses the UI in ``right'' way and that there will not be any system errors. With this in mine, the team will be writing one UI test, per use case.
```

Comment:
...the user uses the UI in *the* right way...

With this in *mind*

Reply:
Oh god, I cannot type :"(